datastore/importers/nmdb-import-rpm/Parser.test.cpp

```diff
@@ -0,0 +1,91 @@
+// =============================================================================
+// Copyright 2022 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MAIN
+#include <boost/test/unit_test.hpp>
+
+#include <netmeld/datastore/parsers/ParserTestHelper.hpp>
+
+#include "Parser.hpp"
+
+namespace nmdp = netmeld::datastore::parsers;
+
+using qi::ascii::blank;
+
+class TestParser : public Parser
+{
+  public:
+    using Parser::start;
+    using Parser::packageLine;
+};
+
+BOOST_AUTO_TEST_CASE(testPackageLine)
+{
+  TestParser tp;
+
+  {
+    const auto& parserRule {tp.packageLine};
+
+    // OK
+    std::vector<std::string> testsOk {
+      "pkg        1.0         all     some description\n",
+      "pkg-other  1.1.99.0-5  amd64   more des-cript -- ion\n",
+      "pkg-o12    1.0+v1      x86_64  description\n",
+    };
+
+    for (const auto& test : testsOk) {
+      BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
+                 "Parse rule 'packageLine': " << test);
+    }
+  }
+}
```

Comment:
This and the `testWhole` are effectively the same.  If you are going to test parts, then it would be best to check the changed state of the object being populated after each individual rule is called.  Otherwise, just remove this and let the test whole suffice.